dispatch/dispatch.go

```diff
@@ -161,20 +167,44 @@ func (d *Dispatcher) run(it provider.AlertIterator) {
 				return
 			}
 
-			level.Debug(d.logger).Log("msg", "Received alert", "alert", alert)
+			// this block is wrapped in a function to make sure that the span
+			// is ended before the next alert is processed
+			func() {
+				traceCtx, span := tracer.Start(d.ctx, "dispatch.Dispatcher.dispatch",
+					trace.WithAttributes(
+						attribute.String("alert.name", alert.Name()),
+						attribute.String("alert.fingerprint", alert.Fingerprint().String()),
+						attribute.String("alert.status", string(alert.Status())),
```

Comment:
I think it should be mentioned that the status of an alert can change between the dispatcher and when the aggregation group is flushed. If that will affect continuation of the trace, we should remove it.

Reply:
> If that will affect continuation of the trace, we should remove it.

Not at all - it's just metadata. But if it can change maybe that's misleading, and if we set it at all we should maybe set it at the end when we have the final status 🤔

Reply:
It can change because the `Status` is calculated using `time.Now()`.